### ISSUE TITLE
[studio] `item` Table won't be updated by pulling Remote Repository

### ISSUE THREAD

**Issue by gozo77** [craftercms/craftercms#4482]
### Describe the bug
We would like to use the functionality a remote repository to a site . 
We can add the remote repository, push and pull but the new folders and files won't be inserted into table `item`. 

### To Reproduce

Create a new authoring site e.g. test-site
Create a repository
Go to http://localhost:8080/studio/site-config?site=test-site#tool/repository
Create  the remote repository to the site
Push the site to repository
Add some folders and files in repository
Pull the repository 

### Expected behavior
The new folders and files should be inserted into table `item`.

### Version
4.0.0

### OS
macOS Catalina
Version 10.15.7

### Browser
Google Chrome
Version 88.0.4324.96

**Comment by gozo77** [comment]
We are using the develop branch version 4.0.0

**Comment by dejan-brkic** [comment]
Already fixed in the other tickets

**Comment by yacdaniel** [comment]
@dejan-brkic please check the following test:
`test a`
1. Create a site from https://github.com/crafter-qa/content_item_tests
2. Add remote https://github.com/crafter-qa/content_item_tests2 to the site and pull
3. Check `item` table and export site data. 

`test b`
1. Create a site from https://github.com/crafter-qa/content_item_tests2
2. Check `item` table  and export site data. 

Compare the files, there are 15 items less in test b. Also in `test b` calling `http://localhost:8080/studio/api/2/content/item_by_path?path=/site/website/newfolder&siteId=sitetestb` returns
```
{
    "response": {
        "code": 0,
        "message": "OK",
        "remedialAction": "",
        "documentationUrl": ""
    },
    "item": null
}
```

**Comment by dejan-brkic** [comment]
@yacdaniel 
This is the list of files that are missing
```
/scripts/new-folder-scripts/.keep
/site/components/new-folder-component/.keep
/site/website/articles/2021/.keep
/site/website/articles/2021/02/.keep
/site/website/articles/2021/02/new-article/.keep
/site/website/new-category-landing/.keep
/site/website/newfolder
/site/website/newfolder/.keep
/static-assets/images/new-image-folder/.keep
/static-assets/item/.keep
/static-assets/item/images/.keep
/static-assets/item/images/2021/.keep
/static-assets/item/images/2021/02/.keep
/static-assets/item/images/2021/02/05/.keep
/templates/web/new-folder-templates/.keep
```
They are all present on the disk but nor in the git index because `.keep` is ignored

**Comment by yacdaniel** [comment]
We should improve how we manage folders + `.keep` files.

**Comment by sumerjabri** [comment]
@dejan-brkic remove .keep from .gitignore so we can shuttle it across git repos. Also, add a field in the bitmap or a column to ignore this file so it can be excluded from queries.

**Comment by dejan-brkic** [comment]
Done
PR:
https://github.com/craftercms/studio/pull/2470

**Comment by yacdaniel** [comment]
Verified fixed in 4.0
